src/framework/uicomponents/view/internal/platform/macos/macospopupviewclosecontroller.mm

```diff
@@ -61,3 +71,18 @@
 
     return false;
 }
+
+void MacOSPopupViewCloseController::initWindowMinimizedObserver()
+{
+    WId wid = parentItem()->window()->winId();
+    NSView* nsView = (__bridge NSView*)reinterpret_cast<void*>(wid);
+    NSWindow* nsWindow = [nsView window];
+
+    minimizeObserverToken = [[NSNotificationCenter defaultCenter]
+                             addObserverForName:@"NSWindowWillMiniaturizeNotification"
+                             object:nsWindow
+                             queue:nil
+                             usingBlock:^(NSNotification*) {
+                                 closeNotification().notify();

```

Comment:
It might be slightly cleaner to make `notifyAboutClose()` a protected method of `PopupViewCloseController`, and use that method instead. For the rest this looks good to me!